bgpd/bgp_evpn.c

```diff
@@ -6144,7 +6144,13 @@ static bool bgp_evpn_remote_ip_hash_cmp(const void *p1, const void *p2)
 	const struct evpn_remote_ip *ip1 = p1;
 	const struct evpn_remote_ip *ip2 = p2;
 
-	return (memcmp(&ip1->addr, &ip2->addr, sizeof(struct ipaddr)) == 0);
+	if (ip1 == NULL && ip2 == NULL)
+		return true;
+
+	if (ip1 == NULL || ip2 == NULL)
+		return false;
+
+	return ip_addr_cmp(&ip1->addr, &ip2->addr);
```

Comment:
There's already `ipaddr_cmp` that properly compares `struct ipaddr`. You implementation is wrong because it only compares the first byte.

Reply:
Fixed.